### ISSUE TITLE
Histogram attributes lost when making a histfactory channel with make_channel

### ISSUE THREAD

**Issue by qbuat** [rootpy/rootpy#668]
I'm experiencing a weird error with the rootpy `make_channel` function. When I try to retrieve histograms from the channel, it seems that the attributes are lost. I've been able to reproduce the problem with this small example:

``` python
from rootpy.stats import histfactory
from rootpy.plotting import Hist

h_data = Hist(10, -1, 1)
h_data.name = 'h_data'
h_data.color = 'black'

h_bkg1 = Hist(10, -1, 1)
h_bkg1.name = 'h_bkg1'
h_bkg1.color = 'red'

h_bkg2 = Hist(10, -1, 1)
h_bkg2.name = 'h_bkg1'
h_bkg2.color = 'blue'

data = histfactory.Data('data')
data.hist = h_data

bkg1 = histfactory.Sample('bkg1')
bkg1.hist = h_bkg1

bkg2 = histfactory.Sample('bkg2')
bkg2.hist = h_bkg2

print 'From the histograms:'
hists = [h_data, h_bkg1, h_bkg2]
for h in hists:
    print h.name, h.color
print
print 'From the samples:'
samples = [data, bkg1, bkg2]
for s in samples:
    print s.name, s.hist.color
print
print 'From the channel:'
channel = histfactory.make_channel('totot', samples[1:], data=samples[0])
for s in [channel.data] + channel.samples:
    print s.name, s.hist.color
```

This gives me:

``` bash
(atlasVE)[qbuat@lhc14 hhana]$ python test_histfactory.py

RooFit v3.60 -- Developed by Wouter Verkerke and David Kirkby
                Copyright (C) 2000-2013 NIKHEF, University of California & Stanford University
                All rights reserved, please read http://roofit.sourceforge.net/license.txt

WARNING:ROOT.TClassTable.Add] class stack<RooAbsArg*,deque<RooAbsArg*> > already in TClassTable
WARNING:ROOT.TUnixSystem.SetDisplay] DISPLAY not set, setting it to 192.168.1.15:0.0
From the histograms:
h_data ('black', 'black', 'black')
h_bkg1 ('red', 'red', 'red')
h_bkg1 ('blue', 'blue', 'blue')

From the samples:
data ('black', 'black', 'black')
bkg1 ('red', 'red', 'red')
bkg2 ('blue', 'blue', 'blue')

From the channel:
data ('black', 'black', 'white')
bkg1 ('black', 'black', 'white')
bkg2 ('black', 'black', 'white')
(atlasVE)[qbuat@lhc14 hhana]$
```

So it looks like someone the decorations are lost or reset to default values.
It's worth mentionning that I see this using ROOT6 (6.02.12-x86_64-slc6-gcc48-opt) but I don't see that happening in ROOT5 (5.34.14-x86_64-slc6-gcc4.8), so maybe this is a problem on the ROOT side.
I'll try to reproduce it without using rootpy but I thought it was worth mentionning here.

Cheers,
Quentin

**Comment by qbuat** [comment]
Quick update: a minimal ROOT-only script seems to indicate that the problem is on the rootpy side.

```python
import ROOT

h_data = ROOT.TH1F('data', '', 10, -1, 1)
h_data.SetLineColor(ROOT.kBlack)

h_bkg1 = ROOT.TH1F('bkg1', '', 10, -1, 1)
h_bkg1.SetLineColor(ROOT.kRed)

h_bkg2 = ROOT.TH1F('bkg2', '', 10, -1, 1)
h_bkg2.SetLineColor(ROOT.kBlue)


data = ROOT.RooStats.HistFactory.Data()
data.SetHisto(h_data)
data.SetName('data')
bkg1 = ROOT.RooStats.HistFactory.Sample('bkg1')
bkg1.SetHisto(h_bkg1)

bkg2 = ROOT.RooStats.HistFactory.Sample('bkg2')
bkg2.SetHisto(h_bkg2)

print 'From the histograms:'
hists = [h_data, h_bkg1, h_bkg2]
for h in hists:
    print h.GetName(), h.GetLineColor()
print
print 'From the samples:'
samples = [data, bkg1, bkg2]
for s in samples:
    print s.GetName(), s.GetHisto().GetLineColor()
print
print 'From the channel:'
channel = ROOT.RooStats.HistFactory.Channel('tototot')
channel.SetData(data)
channel.AddSample(bkg1)
channel.AddSample(bkg2)
for s in channel.GetSamples():
    print s.GetName(), s.GetHisto().GetLineColor()
```
gives me:

```bash
(atlasVE)[qbuat@lhc14 hhana]$ python test_histfactory.py

RooFit v3.60 -- Developed by Wouter Verkerke and David Kirkby
                Copyright (C) 2000-2013 NIKHEF, University of California & Stanford University
                All rights reserved, please read http://roofit.sourceforge.net/license.txt

From the histograms:
data 1
bkg1 632
bkg2 600

From the samples:
data 1
bkg1 632
bkg2 600

From the channel:
bkg1 632
bkg2 600
```

Cheers,
Quentin

**Comment by qbuat** [comment]
Tracking down this a bit more, I think the problem comes from the `asrootpy` in [Line 972 of histfactory.py](https://github.com/rootpy/rootpy/blob/0.8.0/rootpy/stats/histfactory/histfactory.py#L972).

I also noticed that `asrootpy` does not keep the color attributes for samples when doing:

```python
histo_data_test	= asrootpy(super(Channel, self).GetData().GetHisto())
```

However I still can't figure out what the actual issue is and also why is it ROOT6-specific.

**Comment by qbuat** [comment]
Any chance someone has insights on this ?

Thanks!
Quentin

**Comment by ndawe** [comment]
I'm finally here. Sorry.

**Comment by ndawe** [comment]
Fixed in #725 

```
From the histograms:
h_data ('black', 'black', 'black')
h_bkg1 ('red', 'red', 'red')
h_bkg1 ('blue', 'blue', 'blue')

From the samples:
data ('black', 'black', 'black')
bkg1 ('red', 'red', 'red')
bkg2 ('blue', 'blue', 'blue')

From the channel:
data (1, 1, 1)
bkg1 (2, 2, 2)
bkg2 (4, 4, 4)
```
